### PR TITLE
Add gateways command, allowing the retrieval of gateway stats

### PR DESCRIPTION
**Pull request by zaharidichev**
This PR adds a `cluster gateways` command that allows for observing stats of remote cluster gateways as long as they are present in Prometheus. The output of the command looks like: 

```
CLUSTER   NAMESPACE   NAME              NUM SVC   ALIVE   LATENCY_P50   LATENCY_P95   LATENCY_P99
remote    default     another-gateway         1    True         275ms         385ms         397ms
remote    default     linkerd-gateway         2    True         250ms         295ms         299ms
```

Signed-off-by: Zahari Dichev <zaharidichev@gmail.com>

### CODE REVIEW

File: controller/api/public/prometheus.go [hunk 1]
```diff
@@ -176,7 +181,7 @@ func (s *grpcServer) getPrometheusMetrics(ctx context.Context, requestQueryTempl
 	// kick off asynchronous queries: request count queries + 3 latency queries
 	for pt, requestQueryTemplate := range requestQueryTemplates {
 		var query string
-		if pt == promTCPConnections {
+		if pt == promTCPConnections || pt == promGatewayAlive || pt == promNumMirroredServices {
```
Comment (adleong) [inline controller/api/public/prometheus.go:184]: I think this was a bit of a hack to begin with and is getting worse here.  What if we added a boolean `timeWindow` parameter to the signature of `getPrometheusMetrics` that indicates if the query expects a time window?

File: cli/cmd/cluster.go [hunk 2]
```diff
@@ -285,14 +326,169 @@ available across clusters.`,
...
+)
+
+func writeGatewayRowsToBuffer(rows []*pb.GatewaysTable_Row, w *tabwriter.Writer) {
+	maxNameLength := len(gatewayNameHeader)
+	maxNamespaceLength := len(gatewayNamespaceHeader)
+	maxClusterNameLength := len(clusterNameHeader)
+
+	rowsMap := make(map[string]*pb.GatewaysTable_Row)
+
+	for _, r := range rows {
```
[the remark this answers is not in the document]
Comment (kleimkuhler) [reply cli/cmd/cluster.go:386]: Ah cool!

File: controller/api/public/prometheus.go (hunk 1 above)
Comment (zaharidichev) [reply controller/api/public/prometheus.go:184]: I wonder, can we jsut check whether the time window is empty or not? We should always have a default time window if the query needs one, correct ?

File: cli/cmd/cluster.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (zaharidichev) [reply cli/cmd/cluster.go:386]: Wow @adleong that is some real engineering :)

File: controller/api/public/prometheus.go (hunk 1 above)
Comment (adleong) [reply controller/api/public/prometheus.go:184]: makes sense to me

**Pull request merged**